### DISCUSSION TITLE
Gateway is active on the list, but it does not show up on the Checkout page

### DISCUSSION THREAD

**Discussion by Kaji01** [woocommerce/woocommerce#41832] [Checkout Flow]
Title summarizes the issue.  I've been maintaining a payment gateway plugin since 2015 and it's loaded without issues, but for one client yesterday it stopped showing the form on the checkout page, even though the switch said it was active under **WooCommerce > Settings > Payments**.  Instead we just get a message saying that there are now payment methods available.

<img width="1266" alt="image" src="https://github.com/woocommerce/woocommerce/assets/6264427/16b3ca49-94cb-45b3-a925-33d958fb3edc">
<img width="498" alt="image" src="https://github.com/woocommerce/woocommerce/assets/6264427/1b80e6e2-8fab-407d-89b7-49d6da044917">

I've double-checked all of the settings, they're exactly as they should be.  We're using virtual products to test the checkout, as we have for years, and it's not showing.

Not having seen this issue before, I was prepared to write it off as possibly their hosting being configured strangely, but we've tried two clean installs on their hosting and another on my own setup and we're reproducing the issue.

That being the case, I've reviewed the documentation, I've tried rebuilding the plugin from the most recent sample bases I can find, and the result is consistent: I can see the gateway in the list, I can turn it on, but if I go to checkout it says there's nothing there.  Even using the base here.

Here's a list of the action hooks we're tapping into, in case I'm missing one:

```php
add_action('plugins_loaded', 'epn_gateway_init', 11);
add_action('woocommerce_update_options_payment_gateways_' . $this->id, array($this, 'process_admin_options'));
add_filter('plugin_action_links_'.plugin_basename(__FILE__), 'epn_gateway_plugin_links');
```

Has anyone else experienced this?  Is there anything else I can try to get it to appear?

**Comment by wavvves** [comment]
Hi @Kaji01 thank you for reaching out.

From the print screen, I see the client is using the new Checkout Blocks, which is now the new default Checkout experience for new installs. Integrating payments works differently, and you can have a look at how to do that here:

[Payment Method Integration for the Checkout Block](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/third-party-developers/extensibility/checkout-payment-methods/payment-method-integration.md)

If urgency in correcting this issue matters, I suggest reverting to the classic shortcode checkout while you update the integration to work with Checkout blocks.

Was the Checkout page deliberately updated to use Blocks/ was the theme switched recently to a blocks-based theme? 🤔

**Reply by Kaji01** [reply]
These were on new installs, so that would explain things.  I had a sense it had something to do with blocks, but wasn't quite sure how to force it to not use them.  Is it as simple as editing the page to place the shortcode, or is there more to it?

On which note, looks like I'll have to finally figure out React, based on that link...In either case, thank you very much for the tips on what's up!

**Reply by wavvves** [reply]
Correct, you can always edit the page and use the shortcode version to use your current integration.

Glad I could help. Feel free to[ reach out](https://github.com/woocommerce/woocommerce/discussions/new?category=checkout-flow&labels=type%3A+product%20feedback) in case you bump into any problems integrating your payment gateway.
